### PR TITLE
chore(broker): start embedded gateway after cluster services

### LINKED ISSUE #5613: Embedded Gateway is started to early
**Describe the bug**
If we using the embedded gateway then we start it as step 6, but the cluster services later also the topology etc.

This means we accepting already requests at the gateway, but we will not be able to handle them since we have no topology and we are not taking part of the cluster.

This can cause issues like this:

```

Info
2020-10-16 12:03:28.674 CEST
"Starting gateway with configuration { "network" : { "host" : "0.0.0.0", "port" : 26500, "minKeepAliveInterval" : "PT30S" }, "cluster" : { "contactPoint" : "0.0.0.0:26502", "requestTimeout" : "PT15S", "clusterName" : "zeebe-cluster", "memberId" : "gateway", "host" : "zeebe-0.zeebe-broker-service.ae62a34c-3f1b-43c0-a8ff-c03f4e527aa4-zeebe.svc.cluster.local", "port" : 26502, "membership" : { "broadcastUpdates" : false, "broadcastDisputes" : true, "notifySuspect" : false, "gossipInterval" : "PT0.25S", "gossipFanout" : 2, "probeInterval" : "PT1S", "probeTimeout" : "PT2S", "suspectProbes" : 3, "failureTimeout" : "PT10S", "syncInterval" : "PT10S" } }, "threads" : { "managementThreads" : 1 }, "monitoring" : { "enabled" : true, "host" : "0.0.0.0", "port" : 9600 }, "security" : { "enabled" : false, "certificateChainPath" : null, "privateKeyPath" : null }, "longPolling" : { "enabled" : true }, "initialized" : true, "enable" : true }"
Debug
2020-10-16 12:03:29.083 CEST
"Bootstrap Broker-0 [6/11]: embedded gateway started in 418 ms"
Info
2020-10-16 12:03:29.084 CEST
"Bootstrap Broker-0 [7/11]: cluster services"
Error
2020-10-16 12:03:30.082 CEST
"Uncaught exception in 'io.zeebe.gateway.impl.broker.BrokerRequestManager' in phase 'STARTED'. Continuing with next job."
Error
2020-10-16 12:03:30.152 CEST
"Uncaught exception in 'io.zeebe.gateway.impl.broker.BrokerRequestManager' in phase 'STARTED'. Continuing with next job."
Warning
2020-10-16 12:03:33.069 CEST
"Listener io.atomix.raft.roles.FollowerRole$$Lambda$1009/0x0000000100852440@1aeaa3cb not registered"
Debug
2020-10-16 12:03:34.585 CEST
"Bootstrap Broker-0 [7/11]: cluster services started in 5501 ms"
```

Reported in error tool, related error group: https://console.cloud.google.com/errors/CLKTidn42eXorAE?service=zeebe&time=P7D&project=camunda-cloud-240911&authuser=1

**To Reproduce**
Send messages to gateway when broker is not joined.

**Expected behavior**
Accept request after cluster has joined.

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>
```
io.zeebe.gateway.cmd.NoTopologyAvailableException: Expected to pick partition for message with correlation key 'UnsafeBuffer{addressOffset=16, capacity=36, byteArray=[B@4bdcdde9, byteBuffer=null}', but no topology is available
	at io.zeebe.gateway.impl.broker.BrokerRequestManager.determinePartitionIdForPublishMessageRequest(BrokerRequestManager.java:202) ~[zeebe-gateway-0.24.3.jar:0.24.3]
	at io.zeebe.gateway.impl.broker.BrokerRequestManager.determineBrokerNodeIdProvider(BrokerRequestManager.java:169) ~[zeebe-gateway-0.24.3.jar:0.24.3]
	at io.zeebe.gateway.impl.broker.BrokerRequestManager.sendRequestInternal(BrokerRequestManager.java:115) ~[zeebe-gateway-0.24.3.jar:0.24.3]
	at io.zeebe.gateway.impl.broker.BrokerRequestManager.lambda$sendRequestInternal$1(BrokerRequestManager.java:103) ~[zeebe-gateway-0.24.3.jar:0.24.3]
	at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:73) ~[zeebe-util-0.24.3.jar:0.24.3]
	at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) [zeebe-util-0.24.3.jar:0.24.3]
	at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:118) [zeebe-util-0.24.3.jar:0.24.3]
	at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:107) [zeebe-util-0.24.3.jar:0.24.3]
	at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:91) [zeebe-util-0.24.3.jar:0.24.3]
	at io.zeebe.util.sched.ActorThread.run(ActorThread.java:204) [zeebe-util-0.24.3.jar:0.24.3]
```

</p>
</details>

**Environment:**
 - OS: cloud
 - Zeebe Version: seen in 0.24.3
 - Configuration: -

### PR DESCRIPTION
**Pull request by aivinog1**
## Description

I've moved an embedded gateway start step after cluster services start step.

## Related issues

closes #5613 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [X] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: broker/src/main/java/io/zeebe/broker/Broker.java [hunk 1]
```diff
@@ -218,7 +219,6 @@ private StartProcess initStart() {
             return embeddedGatewayService;
           });
     }
-    startContext.addStep("cluster services", () -> atomix.start().join());
     startContext.addStep("topology manager", () -> topologyManagerStep(clusterCfg, localBroker));
```
Comment (Zelldon) [inline broker/src/main/java/io/zeebe/broker/Broker.java:222]: Could we please move it after the topology manger :)

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
Thanks @aivinog1 just a small request.

File: broker/src/main/java/io/zeebe/broker/Broker.java (hunk 1 above)
Comment (aivinog1) [reply broker/src/main/java/io/zeebe/broker/Broker.java]: @Zelldon Done :)

**Review by ChrisKujawa — APPROVED** [review]
Thanks @aivinog1 good job :+1: 

bors r+

**Pull request merged**